pkg/alertmanager/operator.go

```diff
@@ -654,6 +657,20 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	// Check if operator has the permission to read the storage class and if AlertManager Specs is using a Volume Claim.
+	if c.canReadStorageClass && am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != nil && *am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != "" {
+		storageClassName := am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName
+		_, err := c.kclient.StorageV1().StorageClasses().Get(ctx, *storageClassName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				// Error when the incorrect storageClassName is specified.
```

Comment:
(nit) the next line is self-explanatory and doesn't require a comment IMHO
```suggestion
```

Reply:
Done :+1: